README.md

```diff
@@ -73,6 +73,14 @@ The following options are also supported and will be passed directly to the
 - [`lastModified`](https://www.npmjs.com/package/send#lastmodified)
 - [`maxAge`](https://www.npmjs.com/package/send#maxage)
 
+#### `redirect`
+
+Default: `false`
+
+If set to `true`, `fastify-static` redirects to the directory with a trailing slash.
+
+This option is ineffective if `wildcard` is set to `false`.
```

Comment:
They reason why this is not working is with `wildcard: false`, it is because in https://github.com/fastify/fastify-static/blob/master/index.js#L114 we iterate over all files and not over all the directories. You'll need to create routes for every directory you find.

Reply:
`wildcard: false` also ignores the `index` setting, always assuming `'index.html`.  I have a fix for this in the works.

One question I have is which directories should be redirected when `wildcard: false`?  Take `./test/static` served under `http://localhost/static/`.  Should `http://localhost/static/deep` redirect to `http://localhost/static/deep/` knowing that we're not initializing `fastify.get('/static/deep/', ...)`?

Reply:
My next commit expands the scope of this PR beyond my original plans but the `redirect` and `index` options now work when wildcard is disabled.  I also dealt with a couple places where double-slashes were used and fixed a couple issues with redirects that include query strings (`/static?a=b` redirects to `/static/?a=b` instead of losing the query).

Note previously `fastify-static` did call `fastify.get` for sub-directories without the trailing slash, this previously caused a 404 reply.  Now those sub-directories are only registered if `redirect` is enabled and an index file exists.

If needed I can change the logic so that those sub-directories are always registered when `redirect` is enabled, that way we would 301 to the path including the trailing slash.  So `/static/deep` would redirect to `/static/deep/` on the assumption that something else might register a handler for `/static/deep/`.